### DISCUSSION TITLE
"Did you insert it today? What time was it inserted?" - Missing (since update Jul21?)

### DISCUSSION THREAD

**Discussion by Foxy7** [NightscoutFoundation/xDrip#1982] [Q&A]
Hi @Navid200

This is something i mentioned a while ago, but think i've found the point at which it has changed although the solution isn't clear to me.

**Set up**
xDrip Master/Latest Nightly with engineering mode and preemptive restarts
Samsung S7
Dexcom G6 80/81 (1.6.5.25)

**Issue**
The current Master and Nightly builds don't generate the option to back date the sensor start any longer.

I've copied these from the [AndroidAPS xDrip](https://androidaps.readthedocs.io/en/latest/Configuration/xdrip.html#connect-g6-transmitter-for-the-first-time) page, but for current build screens 3 and 4 don't appear for me.

![image](https://user-images.githubusercontent.com/37312469/152600783-c88fc47d-29ab-4d5d-b7c8-06a6ffc67af4.png)
![image](https://user-images.githubusercontent.com/37312469/152601528-b50f3900-2a83-4e98-8616-cc6a546178b4.png)

Yesterday we put on a new sensor and i had to do a hard reset of the TX as i'd replaced the battery, this failed so tried it again at 09:30 before it failed at 09:41, however i wasn't available to check for an hour before i realised it hadn't worked. 

After rebooting the phone the hard reset worked, but when starting the sensor i wasn't given the option (as described above) to set the sensor insertion time back to 9AM. So had to start it at 11AM as though inserted then, as below.

To be clear xDrip took me from screen 2 to screen 5, missing out screenshots 3 and 4.

![image](https://user-images.githubusercontent.com/37312469/152607378-2497acb0-63d6-43ea-8c56-8677820a056c.png)

So i lost 2hrs of sensor data during this (appreciate not the end of the world).

I've had a look at the code and the change appears to be related to this update:

https://github.com/NightscoutFoundation/xDrip/commit/fab3e2bff411efc32d18fe4741703715e9ee3974#

Although i admit the name changes seem fine, but possibly the order of the functions has caused an issue?

Another thought is that the pop up is limited by the firmware, but there doesn't seem to be a line in the code `dexdrip/StartNewSensor.java` checking for the firmware under the `sensorButtonClick` function and it should be fine for the TX i'm using.

Note, I can see this screen if i use an old version of xDrip, e.g. a nightly build from Mar20, but this is no longer a work around as to use AndroidAPS we all need to use the latest Nightly build.

Any help much appreciated.
Thanks.

**Comment by Navid200** [comment]
When using G6 in Native mode, I think it is meaningless to allow the start time to be in the past as the transmitter will not start sending real (not raw) readings until the warm-up is over.
The option was removed on November 27, 2020.
Did it ever work for you?  I don't mean in non-native mode.

**Reply by Foxy7** [reply]
Ah, i see, i'd not thought of it that way, but yes what you say makes sense. 

No I'd not used the time travel before, as I was previously using firefly TXs and only since mid 2021 did i get the rebatteried 80/81 series TXs.

I will run a test later in the month to see how it works with non-native mode and report back.

**Reply by Navid200** [reply]
Please note that using non-native mode will effectively disable factory calibration.  In my opinion, factory calibration is the main advantage of G6 over G5.  I wouldn't give it up myself even if I could.
